src/applications/proxy-rewrite/proxy-rewrite-entry.jsx

```diff
@@ -170,12 +176,95 @@ function shouldActivateInjectedAssets(whitelistItem, proxyRewriteCookieValue) {
   return true;
 }
 
-if (
-  shouldActivateInjectedAssets(
-    getMatchedWhitelistItem(),
-    getProxyRewriteCookieValue(),
-  )
+function getHostnameOverride() {
+  // default to vagovprod
+  const targetEnvironment = (window.location.search.match(
+    /\btargetEnvironment=(\w+)/,
+  ) || [])[1];
+
+  // localhost is not available in the buckets
+  if (targetEnvironment === 'localhost') {
+    return {
+      targetEnvironment,
+      hostnameOverride: 'http://localhost:3001',
+    };
+  }
+
+  // if the bucket is not found, an empty string will use relative paths
+  return {
+    targetEnvironment,
+    hostnameOverride: BUCKETS[targetEnvironment] || '',
+  };
+}
+
+const scriptPaths = [
+  '/js/settings.js',
+  '/generated/polyfills.entry.js',
+  '/generated/vendor.entry.js',
+  '/generated/proxy-rewrite.entry.js',
+];
+
+const linkPaths = ['/generated/styleConsolidated.css'];
+
+function removeInjectedHeaderFooter(
+  links = linkPaths,
+  scripts = scriptPaths,
+  docHead = document.head,
+) {
+  Array.from(docHead.getElementsByTagName('script'))
+    .filter(node => scripts.includes(node.getAttribute('src')))
+    .forEach(node => docHead.removeChild(node));
+
+  Array.from(docHead.getElementsByTagName('link'))
+    .filter(node => links.includes(node.getAttribute('href')))
+    .forEach(node => docHead.removeChild(node));
+}
+
+function addOverrideHeaderFooter(
+  hostname,
+  links = linkPaths,
+  scripts = scriptPaths,
+  docHead = document.head,
 ) {
-  redirectIfNecessary(window);
-  activateInjectedAssets();
+  const docFragment = document.createDocumentFragment();
+  // add <script>s
+  scripts.forEach(path => {
+    const scriptNode = document.createElement('script');
+    scriptNode.setAttribute('src', `${hostname}${path}`);
+
+    docFragment.appendChild(scriptNode);
+  });
+
+  // add <link>
+  links.forEach(path => {
+    const linkNode = document.createElement('link');
+    linkNode.setAttribute('rel', 'stylesheet');
+    linkNode.setAttribute('href', `${hostname}${path}`);
+
+    docFragment.appendChild(linkNode);
+  });
+
+  docHead.appendChild(docFragment);
 }
+
+function main() {
+  // if a build type is passed in the url, then the header for the specific build type is used
+  const { targetEnvironment, hostnameOverride } = getHostnameOverride();
+
+  if (targetEnvironment && targetEnvironment !== currentEnvironment.BUILDTYPE) {
+    removeCurrentHeaderFooter();
+    removeInjectedHeaderFooter();
+    addOverrideHeaderFooter(hostnameOverride);
+  } else if (
+    shouldActivateInjectedAssets(
+      getMatchedWhitelistItem(),
+      getProxyRewriteCookieValue(),
+    )
+  ) {
+    redirectIfNecessary(window);
```

Comment:
The injection code also injects the target version of this file- so the `proxy-rewrite-entry.jsx`. There's no separation of loader / injector with this approach. 

My thinking with this would allow us to test the redirect behavior for specific environment. 
1. I navigate to a teamsite page with `targetEnviroment=vagovstaging`
2. The script see the targetEnvironment and remove / injects the staging assets 
3. The staging assets are run and `redirectIfNecessary()` for the staging environment is used. 

It'll bit a little wonky- in production the redirect trigger before anything is rendered but in testing you'll probably see the page fully render before it redirects. I'll work on testing this.